PanelPal/generate_bed.py

```diff
@@ -1,76 +1,207 @@
+            panel_version,
+            genome_build
+            )
+        variant_validator_api_functions.bedtools_merge(panel_id,
+                                                       panel_version,
+                                                       genome_build
+                                                       )
+        logger.info(
+            "Bedtools merge completed successfully for panel_id=%s",
+            panel_id
+            )
 
-        logger.info("Process completed successfully for panel_id=%s", panel_id)
+        # Log completion of the process
+        logger.info(
+            "Process completed successfully for panel_id=%s",
+            panel_id
+            )
 
     except Exception as e:
-        logger.error("An error occurred in the BED file generation process for panel_id=%s: %s", panel_id, e, exc_info=True)
-        raise  # Reraise the exception after logging it for further handling if needed
+        # Reraise the exception after logging it for further handling if needed
+        logger.error(
+            "An error occurred in the BED file generation process for panel_id=%s: %s",
+            panel_id,
+            e,
+            exc_info=True
+            )
+        raise
+
 
 
 if __name__ == '__main__':
-    """
-    Script entry point: parses command-line arguments and calls the main function.
-    """
+    # Parses command-line arguments and calls the main function.
+
     # Set up argument parsing for the command-line interface (CLI)
-    parser = argparse.ArgumentParser(description='Generate a BED file and perform merge using panel details.')
-    
-    # Define the command-line arguments:
-    # panel_id: The ID of the panel, e.g., "R207"
-    parser.add_argument('-p', '--panel_id', type=str, required=True, help='The ID of the panel, (e.g., "R207").')
+    parser = argparse.ArgumentParser(
+        description='Generate a BED file and perform merge using panel details.'
+        )
 
-    # panel_version: The version of the panel, e.g., "4"
-    parser.add_argument('-v', '--panel_version', type=str, required=True, help='The version of the panel (e.g., "4").')
+    # Define the panel_id argument
+    parser.add_argument('-p', '--panel_id',
+                        type=str,
+                        required=True,
+                        help='The ID of the panel, (e.g., "R207").'
+                        )
+
+    # Define the panel_version argument
+    parser.add_argument('-v', '--panel_version',
+                        type=str,
+                        required=True,
+                        help='The version of the panel (e.g., "4").'
+                        )
+
+    # Define the genome_build argument
+    parser.add_argument('-g', '--genome_build',
+                        type=str,
+                        required=True,
+                        help='The genome build (e.g., GRCh38).',
+                        choices=["GRCh37", "GRCh38"]
+                        )
 
```

Comment:
I think to make this script into something that can be ran with a pip installed version of PanelPal  (e.g. with the command 'panelpal generate-bed -p R21 -v 1.1 -g GRCh38') it might need the argument parser to not be in this part of the code.
So if instead this could be moved to be it's own function that is then called within main(), that might be better for the full installation of panelpal.
I realise this might be a bit of effort if there are any tests for this argument parsing...

Look at check-panel if you need an example of this.

Reply:
Thanks for the example. I have now refactored this. One of the tests needed updating to handle this:

![image](https://github.com/user-attachments/assets/d587dd4e-58ba-4cce-968e-31d06856915d)